### PR TITLE
[quant] Support integer implementations for max_pool2d

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #104226
* __->__ #104225

Summary:
This is needed for representing quantized model in pt2 export quantization flow

Test Plan:
tested by opinfo, python test/test_ops.py

Reviewers:

Subscribers:

Tasks:

Tags:

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

File: aten/src/ATen/native/cpu/MaxPoolKernel.cpp [hunk 1]
```diff
@@ -88,6 +88,42 @@ void cpu_max_pool(
...
+vec::Vectorized<int> isnan<int>(vec::Vectorized<int> vec) {
+  Vectorized<int> ret(false);
+  return ret;
+}
+
+template <>
+vec::Vectorized<int64_t> isnan<int64_t>(vec::Vectorized<int64_t> vec) {
+  Vectorized<int64_t> ret(false);
+  return ret;
+}
```
Comment (jgong5) [inline aten/src/ATen/native/cpu/MaxPoolKernel.cpp:124]: Simplify these with SFINAE `std::is_integral` check?

File: test/quantization/pt2e/test_quantize_pt2e.py [hunk 2]
```diff
@@ -1810,6 +1810,10 @@ def forward(self, input_tensor, hidden_tensor):
             model_graph = convert_pt2e(model_graph)
             self.assertEqual(model_fx(*example_inputs), model_graph(*example_inputs))
 
+    def test_maxpool2d_int8(self):
+        x = torch.randint(1, 100, (1, 3, 3, 3), dtype=torch.int8)
```
Comment (jgong5) [inline test/quantization/pt2e/test_quantize_pt2e.py:1814]: Is `torch.uint8` a possible input too?

File: aten/src/ATen/native/cpu/MaxPoolKernel.cpp (hunk 1 above)
Comment (jerryzh168) [reply aten/src/ATen/native/cpu/MaxPoolKernel.cpp:124]: good point, thanks

File: test/quantization/pt2e/test_quantize_pt2e.py (hunk 2 above)
Comment (jerryzh168) [reply test/quantization/pt2e/test_quantize_pt2e.py:1814]: yes, this is just a sanity check, I'm not sure where should I add test yet

File: aten/src/ATen/native/cpu/MaxPoolKernel.cpp [hunk 3]
```diff
@@ -471,7 +481,7 @@ void max_pool2d_kernel_impl(
       break;
     }
     case at::MemoryFormat::ChannelsLast: {
-      AT_DISPATCH_FLOATING_TYPES_AND(ScalarType::BFloat16, input.scalar_type(), "max_pool2d_channels_last", [&] {
+      AT_DISPATCH_ALL_TYPES_AND(ScalarType::BFloat16, input.scalar_type(), "max_pool2d_channels_last", [&] {
```
Comment (kimishpatel) [inline aten/src/ATen/native/cpu/MaxPoolKernel.cpp:484]: is complex part of ALL_TYPES?

File: aten/src/ATen/native/cpu/MaxPoolKernel.cpp (hunk 1 above)
Comment (jerryzh168) [reply aten/src/ATen/native/cpu/MaxPoolKernel.cpp]: looks like it does not work somehow, I'll need to revert this

File: aten/src/ATen/native/cpu/MaxPoolKernel.cpp (hunk 3 above)
Comment (jerryzh168) [reply aten/src/ATen/native/cpu/MaxPoolKernel.cpp:484]: no, all_types == floating_point types + integral types: https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/Dispatch.h#L378

**Review by kimishpatel — APPROVED** [review]
Looks good. Did you test via python/test/test_ops.py?

**Comment by jerryzh168** [comment]
> Looks good. Did you test via python/test/test_ops.py?

yeah I ran the test locally, there are some test failures unrelated to the PR, maybe they are related to test environment, I think it should be OK to just rely on the OSSCI here

File: test/onnx/test_fx_op_consistency.py [hunk 4]
```diff
@@ -447,6 +447,11 @@
         "nn.functional.embedding",
         reason=onnx_test_common.reason_onnx_script_does_not_support("aten.embedding_renorm.default"),
     ),
+    xfail(
```
Comment (jerryzh168) [inline test/onnx/test_fx_op_consistency.py:450]: cc @BowenBao @abock @thiagocrepaldi FYI

**Pull request closed without merging**